File: python3_solution_set2/2410. Maximum Matching of Players With Trainers.py
import bisect
class Solution:
    def matchPlayersAndTrainers(self, players: List[int], trainers: List[int]) -> int:
        
        ans = 0
        players = sorted(players)
        trainers = [-1] + sorted(trainers)
        i = 0
        t = len(trainers)
        for x in players:
            done = False
            while i < t and x > trainers[i]:
                i+=1
                
            if i < t and x <= trainers[i]:
                ans+=1
                i+=1
                
        return ans
                
            
        

File: cpp_solutions/2410. Maximum Matching of Players With Trainers.cpp
class Solution {
public:
    int matchPlayersAndTrainers(vector<int>& players, vector<int>& trainers) {
        int ans = 0;
        sort(players.begin(), players.end());
        
        
        sort(trainers.begin(), trainers.end());
        int i = 0;
        int t = trainers.size();
        for(auto it : players) {
            
            while (i < t && it > trainers[i]) {
                ++i;
            }
            
            if (i < t && it <= trainers[i]) {
                ++i;
                ++ans;
            }
            
        }
        return ans;
    }
};
